src/codegen/mod.rs

```diff
@@ -1007,8 +1011,10 @@ impl CodeGenerator for CompInfo {
             };
 
             let mut attrs = vec![];
-            if let Some(comment) = field.comment() {
-                attrs.push(attributes::doc(comment));
+            if ctx.options().generate_comments {
```

Comment:
This is a whole lot of copy paste... What if we made `Item::comment` and `Field::comment` take the context and check themselves? The earlier we push this check, we could even potentially avoid keeping the comments around in memory.

I would be fine with landing this code now and investigating this later, but if you do want to land the code now, please file an issue and maybe mark it as a good first bug to attract new contributors.